esp32: Change the sendbreak duration to >= 15 bit times.

robert-hh opened the pull request:
It used to be 10 bit times, which is too short. The break state must be longer than a regular character time, at least 13 bit times. May be longer. This is implemented by changing the baud rate for sendbreak() to 2/3 of what has been configured.

Tested with a GENERIC_SPIRAM, GENERIC_C3 and UM_TINYS2 board.

robert-hh committed:
esp32: Change sendbreak time to 15 bit times.

It used to be 10 bit times, which is too short. The break state must be
longer than a regular character time, at least 13 bit times. May be
longer. This is implemented by rfeduing the baud rate to 2/3 of what has
been configured.

Tested with a GENERIC_SPIRAM, GENERIC_C3 and UM_TINYS2 board.

dpgeorge commented on ports/esp32/machine_uart.c:
```diff
@@ -414,13 +416,15 @@ STATIC mp_obj_t machine_uart_sendbreak(mp_obj_t self_in) {
         // Cannot synthesise break
         mp_raise_OSError(MP_EPERM);
     }
```
I think this set of if-statements can now be removed, it should be enough to just decrease the baudrate.

robert-hh replied:
That's true.

robert-hh replied:
Updated.

dpgeorge replied:
Looks good!

robert-hh committed:
esp32: Simplify the sendbreak function.

The break time will now vary with data length and parity setting, but
will at least be 15 bit times.

dpgeorge commented:
Merged in 13dceaa4ea4336eddb763985c2274cfa497c8e82
